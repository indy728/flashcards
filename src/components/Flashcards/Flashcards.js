import React, { Component } from 'react'
import styled from 'styled-components'
import { connect } from 'react-redux'
import * as actions from '../../store/actions'
import Flashcard from './Flashcard/Flashcard'
import FlashcardControls from './FlashcardControls/FlashcardControls'

const Wrapper = styled.div`
    width: 100%;
    height: 50rem;
    max-height: 100%;
    /* z-index: 500; */
    top: 50%;
    left: 0%;
    transform: translateY(-50%);
    display: flex;
`

const SlideshowFeature = styled.div`
    flex: 1;
    height: 100%;

    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;

    /* > * {
        transform: ${props => `translateX(calc(-100% * ${props.index}))`};
    } */
    /* background-color: orange; */
`

class Flashcards extends Component {

    render() {
        const { cocktails, stack, flashcards } = this.props
        const flashcardArray = []

        stack.pool.forEach(id => {
            flashcardArray.push(
                <Flashcard
                    key={id}
                    cocktail={cocktails[id]}
                    />
            )
        })
        const featureFlashcard = flashcardArray[flashcards.flashcardsIndex]

        return (
            <Wrapper
                className='slideshow'
                >
                <FlashcardControls
                    className='slideshow--control-back'
                    buttonSymbol='chevron-left'
                    clicked={() => this.props.incrementFlashcardIndex(-1)}
                    />
                <SlideshowFeature>
                    {featureFlashcard}
                </SlideshowFeature>
                <FlashcardControls
                    className='slideshow--control-forward'
                    buttonSymbol='chevron-right'
                    clicked={() => this.props.incrementFlashcardIndex(1)}
                    />
            </Wrapper>
        )
    }
}

const mapStateToProps = state => {
    return {
        cocktails: state.cocktails.cocktails,
        stack: state.learning.stack,
        flashcards: state.learning.flashcards
    }
}

const mapDispatchToProps = dispatch => {
    return {
        incrementFlashcardIndex: i => dispatch(actions.incrementFlashcardIndex(i))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Flashcards)
