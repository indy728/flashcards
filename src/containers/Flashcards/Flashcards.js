import React, { Component } from 'react'
import styled from 'styled-components'
import Flashcard from '../../components/Flashcard/Flashcard'

const Wrapper = styled.div`
    /* width: 50rem; */
    padding: 5rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`

const initFlashcards = [
    {
        name: 'The Keeper',
        ingredients: [
            {
                name: 'vodka',
                qty: '1 1/2 oz',
            },
            {
                name: 'aperol',
                qty: '1/2 oz',
            },
            {
                name: 'honey syrup',
                qty: '1/2 oz',
            },
            {
                name: 'lemon juice',
                qty: '3/4 oz',
            },
        ],
        instructions: 'shake, strain, top with prosecco',
        glassware: 'coupe',
        garnish: 'lemon wheel'
    },
    // {
    //     name: 'Opening Statement',
    //     ingredients: [
    //         {
    //             name: 'gin',
    //             qty: '3/4 oz',
    //         },
    //         {
    //             name: "Kina L'Aero D'Or",
    //             qty: '3/4 oz',
    //         },
    //         {
    //             name: 'luxardo',
    //             qty: '3/4 oz',
    //         },
    //         {
    //             name: 'lemon juice',
    //             qty: '3/4 oz',
    //         },
    //         {
    //             name: 'orange bitters',
    //             qty: '2 dashes',
    //         },
    //     ],
    //     instructions: 'shake and strain',
    //     glassware: 'coupe',
    //     garnish: 'none'
    // },
]

class Flashcards extends Component {
    state = {
        cocktails: []
    }

    componentDidMount() {
        this.setState({cocktails: initFlashcards})
    }

    render() {
        const { cocktails } = this.state
        const flashcards = cocktails.map(cocktail => {
            return (
                <Flashcard 
                    key={cocktail.name}
                    cocktail={cocktail}
                    />
            )
        })

        console.log(initFlashcards)
        return (
            <Wrapper className='flashcards'>
                {flashcards}
            </Wrapper>
        )
    }
}

export default Flashcards
