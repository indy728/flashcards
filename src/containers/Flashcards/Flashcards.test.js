import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Flashcards from './Flashcards'

configure({adapter: new Adapter()})

describe('Flashcards', () => {
    const flashcards = shallow(<Flashcards />)

    it('renders correctly', () => {
        expect(flashcards).toMatchSnapshot()
    })

    // TEST THAT STATE INITIALIZES FLASHCARD ARRAY FROM PROPS

    // TEST THAT COUNT OF FLASHCARD COMPONENTS RENDERED IS  EQUAL TO THE LENGTH OF STATE.FLASHCARDS
})