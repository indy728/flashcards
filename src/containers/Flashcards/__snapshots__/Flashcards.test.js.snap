// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flashcards renders correctly 1`] = `ShallowWrapper {}`;
